internal/services/azurestackhci/stack_hci_cluster_resource.go

```diff
@@ -145,6 +188,13 @@ func resourceArmStackHCIClusterRead(d *pluginsdk.ResourceData, meta interface{})
 		if props := model.Properties; props != nil {
 			d.Set("client_id", props.AadClientId)
 			d.Set("tenant_id", props.AadTenantId)
+			if v, ok := d.GetOk("automanage_configuration_id"); ok {
+				automanageConfigId, err := autoParse.AutomanageConfigurationID(v.(string))
+				if err != nil {
+					return err
+				}
+				d.Set("automanage_configuration_id", automanageConfigId.ID())
```

Comment:
I think we should be performing a `hciAssignmentClient.Get` here, parsing the automanage config ID insensitively and setting that into state instead of using the value provided in the config?

Reply:
Agreed, let me update the code today.